### PR TITLE
feat(Teams): backport support for teams

### PR DESCRIPTION
**Pull request by SpaceEEC**
**Please describe the changes this PR makes and why it should be merged:**
This PR backports #3350 (a22aabf6a827f8643cf84939e9065b6c37b289e0) in a semver-minor manner.
(Namingly adding a `team` property opposed to replacing `owner`)

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/OAuth2Application.js [hunk 1]
```diff
@@ -103,6 +104,14 @@ class OAuth2Application {
        */
       this.owner = this.client.dataManager.newUser(data.owner);
     }
+
+    if (data.team) {
+      /**
+       * The owning team of this OAuth application
+       * @type {?Team}
+       */
+      this.team = new Team(this.client, data.team);
```
Comment (bdistin) [inline src/structures/OAuth2Application.js]: The team property should always exist as null or Team, not excluding the property if no data.team, so the object keeps consistent shape.

Comment (bdistin) [reply src/structures/OAuth2Application.js]: Well, and also, it should be deprecated since it will be gone in v12.

Comment (SpaceEEC) [reply src/structures/OAuth2Application.js]: How would such a deprecation look like? Emitting a process warning on access seems unjustified as there is no way to fix this before v12. Maybe a warn or info?

Comment (bdistin) [reply src/structures/OAuth2Application.js]: At minimum just a docstring @deprecated. But probably that with an explanation (in a warn or info tag) in the description.

**Pull request merged**